providers/provider_data.go

```diff
@@ -26,16 +26,16 @@ type ProviderData struct {
 // Data returns the ProviderData
 func (p *ProviderData) Data() *ProviderData { return p }
 
-func (p *ProviderData) ClientSecret() (ClientSecret string, err error) {
-	if p.ClientSecret_ != "" || p.ClientSecretFile == "" {
-		return p.ClientSecret_, nil
+func (p *ProviderData) GetClientSecret() (ClientSecret string, err error) {
+	if p.ClientSecret != "" || p.ClientSecretFile == "" {
+		return p.ClientSecret, nil
 	}
 
 	// Getting ClientSecret can fail in runtime so we need to report it without returning the file name to the user
-	ClientSecret_, err := ioutil.ReadFile(p.ClientSecretFile)
+	FileClientSecret, err := ioutil.ReadFile(p.ClientSecretFile)
```

Comment:
Nit, this should be lowercased
```suggestion
	fileClientSecret, err := ioutil.ReadFile(p.ClientSecretFile)
```